syncer/sharding_group.go

```diff
@@ -93,23 +94,35 @@ type ShardingGroup struct {
 	// (0, len(sources)): waiting for syncing
 	// NOTE: we can make remain to be configurable if needed
 	remain       int
-	sources      map[string]bool     // source table ID -> whether source table's DDL synced
-	sourceDDLs   map[string][]string // source table ID -> ddl text; detect multiple ddl for on table in a sharding ddl
-	IsSchemaOnly bool                // whether is a schema (database) only DDL TODO: zxc add schema-level syncing support later
-	firstPos     *mysql.Position     // first DDL's binlog pos, used to restrain the global checkpoint when un-resolved
-	firstEndPos  *mysql.Position     // first DDL's binlog End_log_pos, used to re-direct binlog streamer after synced
-	ddls         []string            // DDL which current in syncing
+	sources      map[string]bool // source table ID -> whether source table's DDL synced
+	IsSchemaOnly bool            // whether is a schema (database) only DDL TODO: zxc add schema-level syncing support later
+
+	sourceID        string                  // associate dm-worker source ID
+	shardmetaSchema string                  // shard meta storage schema name
+	shardmetaTable  string                  // shard meta storage table name
+	meta            *shardmeta.ShardingMeta // sharding sequence meta storage
+
+	firstPos    *mysql.Position // first DDL's binlog pos, used to restrain the global checkpoint when un-resolved
+	firstEndPos *mysql.Position // first DDL's binlog End_log_pos, used to re-direct binlog streamer after synced
+	ddls        []string        // DDL which current in syncing
 }
 
 // NewShardingGroup creates a new ShardingGroup
-func NewShardingGroup(sources []string, isSchemaOnly bool) *ShardingGroup {
+func NewShardingGroup(cfg *config.SubTaskConfig, sources []string, meta *shardmeta.ShardingMeta, isSchemaOnly bool) *ShardingGroup {
 	sg := &ShardingGroup{
-		remain:       len(sources),
-		sources:      make(map[string]bool, len(sources)),
-		sourceDDLs:   make(map[string][]string),
-		IsSchemaOnly: isSchemaOnly,
-		firstPos:     nil,
-		firstEndPos:  nil,
+		remain:          len(sources),
+		sources:         make(map[string]bool, len(sources)),
+		IsSchemaOnly:    isSchemaOnly,
+		sourceID:        cfg.SourceID,
+		shardmetaSchema: cfg.MetaSchema,
+		shardmetaTable:  fmt.Sprintf(shardmeta.MetaTableFormat, cfg.Name),
```

Comment:
How about passing `shardmetaSchema`/`shardmetaTable` in from `ShardingGroupKeeper` (only do the `fmt` in `ShardingGroupKeeper`)? 
and even we can make `sourceID` as an argument (no `config.SubTaskConfig` need to know for `ShardingGroup`)?